src/main/io/osd.c

```diff
@@ -123,69 +123,63 @@ static displayPort_t *osd7456DisplayPort;
 
 PG_REGISTER_WITH_RESET_FN(osdConfig_t, osdConfig, PG_OSD_CONFIG, 0);
 
-/**
- * Converts altitude/distance based on the current unit system (cm or 1/100th of ft).
- * @param alt Raw altitude/distance (i.e. as taken from baro.BaroAlt)
- */
-static int32_t osdConvertDistanceToUnit(int32_t dist)
-{
-    switch (osdConfig()->units) {
-        case OSD_UNIT_IMPERIAL:
-            return (dist * 328) / 100; // Convert to feet / 100
-        default:
-            return dist;               // Already in meter / 100
-    }
-}
+#define OSD_UNIT_DIST osdConfig()->units
+#define OSD_UNIT_DIST_ICON(unit) (((unit) == OSD_UNIT_METRIC) ? SYM_DISTHOME_M : SYM_DISTHOME_FT)
+
+#define OSD_UNIT_ALT osdConfig()->units
+#define OSD_UNIT_ALT_ICON(unit) (((unit) == OSD_UNIT_METRIC) ? SYM_ALTM : SYM_ALTFT)
+
+#define OSD_UNIT_VEL osdConfig()->units
 
 /**
  * Converts altitude/distance into a string based on the current unit system.
  * @param alt Raw altitude/distance (i.e. as taken from baro.BaroAlt in centimeters)
+ * @param showUnit Show unit behind numeric value
  */
-static void osdFormatDistanceStr(char* buff, int32_t dist)
+static void osdFormatDistanceStr(char* buff, int32_t dist, bool showUnit)
 {
-	int32_t dist_abs = abs(osdConvertDistanceToUnit(dist));
-
-    switch (osdConfig()->units) {
+    int32_t dist_abs = abs(dist);
+    switch (OSD_UNIT_DIST) {
```

Comment:
@martinbudden My thought was to make it easier to implement different units for distance/altitude and speed (see #1756). I can change it back to the version without macro if that is preferable.